lib/widgets/breez_avatar_dialog.dart

```diff
@@ -11,150 +13,249 @@ import 'package:image/image.dart' as DartImage;
+                          );
+                        }
+                        return Padding(
+                          child: AspectRatio(
+                              aspectRatio: 1,
+                              child: BreezAvatar(
+                                _pickedImage?.path ?? snapshot.data.avatarURL,
+                                radius: 36.0,
+                              )),
+                          padding: EdgeInsets.only(top: 26.0),
+                        );
+                      }
+                    },
+                  ),
+                  Expanded(
+                    child: FlatButton(
+                      padding: EdgeInsets.only(
+                        bottom: 20.0,
+                        top: 26.0,
+                      ),
+                      child: AutoSizeText(
+                        'GALLERY',
+                        style: theme.whiteButtonStyle,
+                        maxLines: 1,
+                        minFontSize: MinFontSize(context).minFontSize,
+                        stepGranularity: 0.1,
+                        group: _autoSizeGroup,
+                      ),
+                      onPressed: () async {
+                        await _pickImage().then(
+                          (file) {
+                            setState(() {
+                              _pickedImage = file;
+                            });
+                          },
+                        );
+                      },
+                    ),
+                  ),
+                ],
               ),
             ),
+          ]),
+          content: SingleChildScrollView(
+            child: ListBody(
+              children: <Widget>[
+                Theme(
+                  data: ThemeData(
+                      primaryColor:
+                          Theme.of(context).primaryTextTheme.body1.color,
+                      hintColor:
+                          Theme.of(context).primaryTextTheme.body1.color),
+                  child: TextField(
+                      style: Theme.of(context).primaryTextTheme.body1,
+                      controller: _nameInputController,
+                      decoration: InputDecoration(hintText: 'Enter your name'),
+                      onSubmitted: (text) {}),
+                )
+              ],
+            ),
+          ),
+          actions: <Widget>[
+            FlatButton(
+              child: Text('CANCEL',
+                  style: Theme.of(context).primaryTextTheme.button),
+              onPressed: _state == UPLOAD_STATE.UPLOAD_IN_PROGRESS
+                  ? null
+                  : () {
+                      Navigator.of(context).pop();
+                    },
+            ),
+            FlatButton(
+              child: Text('SAVE',
+                  style: Theme.of(context).primaryTextTheme.button),
+              onPressed: () async {
+                if (_pickedImage != null) {
+                  await _pickedImage
```

Comment:
It will be nice to refactor the  upload to its own function that returns a future and encapsulate all the details (reading, scaling).
It currently hard to follow the code and I think there are some bugs like if the upload fails it still update the user name, etc...

Reply:
It was deliberate but I guess you're right, it should save the changes as a whole.

Reply:
[Done](https://github.com/breez/breezmobile/pull/283/commits/40f99d4f9d8709f7de4dfee8872f13189efa6093)